cardano-cli/src/Cardano/CLI/EraBased/Options/Governance/Poll.hs

```diff
@@ -1,73 +1,75 @@
+{-# LANGUAGE GADTs #-}
+
 module Cardano.CLI.EraBased.Options.Governance.Poll
   ( pGovernancePollCmds,
+    pGovernanceAnswerPoll, {- Exported for legacy commands -}
+    pGovernanceCreatePoll, {- Exported for legacy commands -}
+    pGovernanceVerifyPoll  {- Exported for legacy commands -}
   )
 where
 
 import           Cardano.Api
 
 import           Cardano.CLI.EraBased.Commands.Governance.Poll (GovernancePollCmds (..))
 import           Cardano.CLI.EraBased.Options.Common
-import           Cardano.Prelude (catMaybes, isInfixOf)
 
-import           Control.Monad (when)
 import           Data.Foldable
 import           Options.Applicative hiding (help, str)
 import qualified Options.Applicative as Opt
 
+-- | Era-dependent entry point. Only returns @Just _@ if the era
+-- is babbage, since those commands are babbage-specific.
+-- Use the parser of individual commands to avoid providing the era.
+-- See https://github.com/input-output-hk/cardano-cli/issues/340 for context.
 pGovernancePollCmds :: ()
   => CardanoEra era
-  -> Maybe (Parser (GovernancePollCmds era))
+  -> Maybe (Parser GovernancePollCmds)
 pGovernancePollCmds era =
-  case parsers of
-    [] -> Nothing
-    _ -> Just $ asum parsers
+  case era of
+    ByronEra -> Nothing
+    ShelleyEra -> Nothing
+    AllegraEra -> Nothing
+    MaryEra    -> Nothing
+    AlonzoEra  -> Nothing
+    BabbageEra -> Just $ asum parsers
+    ConwayEra  -> Nothing
   where
     parsers =
-      catMaybes
         [ subParser "create-poll"
-            <$> ( Opt.info
-                    <$> pGovernanceCreatePoll era
-                    <*> pure (Opt.progDesc "Create an SPO poll")
-                ),
+            ( Opt.info
+                pGovernanceCreatePoll
+                (Opt.progDesc "Create an SPO poll")
+             ),
           subParser "answer-poll"
-            <$> ( Opt.info
-                    <$> pGovernanceAnswerPoll era
-                    <*> pure (Opt.progDesc "Answer an SPO poll")
-                ),
+            ( Opt.info
+                pGovernanceAnswerPoll
+                (Opt.progDesc "Answer an SPO poll")
+            ),
           subParser "verify-poll"
-            <$> ( Opt.info
-                    <$> pGovernanceVerifyPoll era
-                    <*> pure (Opt.progDesc "Verify an answer to a given SPO poll")
-                )
+            ( Opt.info
+                pGovernanceVerifyPoll
+                (Opt.progDesc "Verify an answer to a given SPO poll")
+            )
```

Comment:
The previous implementation was correct.

Reply:
Reverted :+1: